include/CL/opencl.hpp

```diff
@@ -2396,35 +2385,39 @@ class Device : public detail::Wrapper<cl_device_id>
                 cl_uint * /*num_devices*/ ) CL_API_SUFFIX__VERSION_1_1;
 
         static PFN_clCreateSubDevicesEXT pfn_clCreateSubDevicesEXT = NULL;
-        CL_HPP_INIT_CL_EXT_FCN_PTR_(clCreateSubDevicesEXT);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(platform, clCreateSubDevicesEXT);
 
-        cl_uint n = 0;
-        cl_int err = pfn_clCreateSubDevicesEXT(object_, properties, 0, NULL, &n);
-        if (err != CL_SUCCESS) {
-            return detail::errHandler(err, __CREATE_SUB_DEVICES_ERR);
-        }
+        if (pfn_clCreateSubDevicesEXT) {
+            cl_uint n = 0;
+            error = pfn_clCreateSubDevicesEXT(object_, properties, 0, NULL, &n);
+            if (error != CL_SUCCESS) {
+                return detail::errHandler(error, __CREATE_SUB_DEVICES_ERR);
+            }
 
-        vector<cl_device_id> ids(n);
-        err = pfn_clCreateSubDevicesEXT(object_, properties, n, ids.data(), NULL);
-        if (err != CL_SUCCESS) {
-            return detail::errHandler(err, __CREATE_SUB_DEVICES_ERR);
-        }
-        // Cannot trivially assign because we need to capture intermediates 
-        // with safe construction
-        if (devices) {
-            devices->resize(ids.size());
+            vector<cl_device_id> ids(n);
+            error = pfn_clCreateSubDevicesEXT(object_, properties, n, ids.data(), NULL);
+            if (error != CL_SUCCESS) {
+                return detail::errHandler(error, __CREATE_SUB_DEVICES_ERR);
+            }
 
-            // Assign to param, constructing with retain behaviour
-            // to correctly capture each underlying CL object
-            for (size_type i = 0; i < ids.size(); i++) {
-                // We do not need to retain because this device is being created 
-                // by the runtime
-                (*devices)[i] = Device(ids[i], false);
+            // Cannot trivially assign because we need to capture intermediates
+            // with safe construction
+            if (devices) {
+                devices->resize(ids.size());
+
+                // Assign to param, constructing with retain behaviour
+                // to correctly capture each underlying CL object
+                for (size_type i = 0; i < ids.size(); i++) {
+                    // We do not need to retain because this device is being created 
+                    // by the runtime
+                    (*devices)[i] = Device(ids[i], false);
+                }
```

Comment:
In the spirit of no raw loops. Mild bikeshedding, feel free to discard if more trouble than use.
```suggestion
                devices->reserve(ids.size());

                // Assign to param, constructing with retain behaviour
                // to correctly capture each underlying CL object
                std::transform(ids.cbegin(), ids.cend(), std::back_inserter(devices), [](const cl_device_id& id) {
                    // We do not need to retain because this device is being created 
                    // by the runtime
                    return Device(id, false);
                });
```

Reply:
Mind you, this requires `algorithm`. `iterator` is already included.